libs/sdk-ui-filters/src/DateFilter/DateFilterBody/DateFilterBodyButton.tsx

```diff
@@ -8,12 +8,14 @@ interface IDateFilterBodyButtonProps {
     messageId: string;
     className: string;
     disabled?: boolean;
+    id?: string;
 }
 
 const DateFilterBodyButtonComponent: React.FC<IDateFilterBodyButtonProps & WrappedComponentProps> = (
     props,
 ) => (
     <Button
+        id={!props.disabled && props.id}
```

Comment:
detecting enabled and disabled state of Apply button by this ID presence detection seems to be really fragile. Why not to use state of dateFilter itself. Disabled states of these buttons are depending only on known errors and so on

Reply:
Removed this, check Apply button availability by errors as suggested.